components/framework-axp/src/main/java/framework/cache/AXPCache.java

```diff
@@ -0,0 +1,269 @@
+package framework.cache;
+
+import java.io.IOException;
+import java.lang.ref.SoftReference;
+import java.util.Optional;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Cache implementation for the AXP.
+ * The cache use a {@link ConcurrentHashMap} for store data in memory. The cache has time-to-live and a purge
+ * options in side the implementation. Also this has the support for init cache with no time limit.
+ * This has the capability of the ttl configuration support in object level as well.
+ * <p>
+ * Note that in the implementation {@link SoftReference} has used to wrap the object.
+ * softly-reachable objects are guaranteed to have been cleared before the virtual machine throws an OutOfMemoryError.
+ * <p>
+ * !!! HIGH IMPORTANT !!!
+ * <p>
+ * All the objects stored in the cache will be return without cloning or deep cloning due to improve the cache performance.
+ * Because of that user should be responsible of the mutability of the stored object.
+ */
+public final class AXPCache implements ICache
```

Comment:
Occurred that it "may" be possible to support this implementation through javax.cache, by registering AXPCache as a separate cache manager. Will find more on this and confirm. Nothing to fix on this yet. This comment is to serve as just a note.

Reply:
Noted